File: python/backtracking/subsets.py
class Solution:
    def subsets(self, nums):
        res, sub = [], []
        def dfs(index):
            if index == len(nums):
                res.append(sub.copy())

            sub.append(nums[index])
            dfs(index+1)

            sub.pop()
            dfs(index+1)

        dfs(0)
        return res

File: js/src/backtracking/subsets.js
const subsets = (nums) => {
  const res = [];
  const sub = [];
  const dfs = (index) => {
    if (index === nums.length) {
      res.push(...sub);
      return;
    }
    sub.push(nums[index]);
    dfs(index + 1);

    sub.pop();
    dfs(index + 1);
    return;
  };
  return dfs(0);
};
